### PR TITLE
Fix removing temporary directories

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
If removing the whole temporary directory fails, remove only its contents.

### CODE REVIEW

File: utils/io/fileutils/files.go [hunk 1]
```diff
@@ -708,3 +708,29 @@ func MoveFile(sourcePath, destPath string) (err error) {
 	err = os.Remove(sourcePath)
 	return errorutils.CheckError(err)
 }
+
+// RemoveDirContents removes the contents of the directory, without removing the directory itself.
+// If it encounters an error before removing all the files, it stops and returns that error.
+func RemoveDirContents(dirPath string) (err error) {
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go:714]: RemoveDirContents --> RemoveDirContent

File: utils/io/fileutils/files.go [hunk 2]
```diff
@@ -708,3 +708,29 @@ func MoveFile(sourcePath, destPath string) (err error) {
 	err = os.Remove(sourcePath)
 	return errorutils.CheckError(err)
 }
+
+// RemoveDirContents removes the contents of the directory, without removing the directory itself.
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go:712]: RemoveDirContents --> RemoveDirContent
contents --> content

File: utils/io/fileutils/files_test.go [hunk 3]
```diff
@@ -247,5 +247,29 @@ func TestGetFileAndDirFromPath(t *testing.T) {
 		assert.Equal(t, test.expectedFile, File, "Wrong file name for path: "+test.path)
 		assert.Equal(t, test.expectedDir, Dir, "Wrong dir for path: "+test.path)
 	}
+}
+
+func TestRemoveDirContents(t *testing.T) {
```
Comment (eyalbe4) [inline utils/io/fileutils/files_test.go:252]: TestRemoveDirContents --> TestRemoveDirContent

File: utils/io/fileutils/temp.go [hunk 4]
```diff
@@ -50,10 +50,16 @@ func RemoveTempDir(dirPath string) error {
...
+	if !exists {
+		return nil
 	}
-	return nil
+	err = os.RemoveAll(dirPath)
+	if err == nil {
+		return nil
+	}
+	// Sometimes removing the directory fails (usually in Windows in GitHub Actions) because it's locked by another process.
+	// In this case, we'll only remove the contents of the directory, and let CleanOldDirs() remove the directory itself at a later time.
```
Comment (eyalbe4) [inline utils/io/fileutils/temp.go:61]: contents --> content

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/io/fileutils/files.go (hunk 1 above)
Comment (asafgabai) [reply utils/io/fileutils/files.go:714]: From what I saw on the internet, for this meaning, "contents" is better than "content".

**Pull request merged**